sortinghat/core/schema.py

```diff
@@ -725,6 +727,25 @@ def resolve_individuals(self, info, filters=None,
 
         if filters and 'uuid' in filters:
             query = query.filter(mk=filters['uuid'])
+        if filters and 'term' in filters:
+            search_term = filters['term']
+
+            # Perform search within identities
+            query_uids = Identity.objects.order_by('uuid')
+            results = query_uids.filter(Q(name__icontains=search_term) |
+                                        Q(email__icontains=search_term) |
+                                        Q(username__icontains=search_term))
+            result_uids = [result.individual.mk for result in results]
+
+            # Perform search within profiles
+            query_profiles = Profile.objects.order_by('individual')
+            results = query_profiles.filter(Q(name__icontains=search_term) |
+                                            Q(email__icontains=search_term))
+            result_profiles = [result.individual.mk for result in results]
+
+            # Filter matching individuals by their mk
+            query = query.filter(Q(mk__in=result_uids) |
+                                 Q(mk__in=result_profiles))
```

Comment:
Isn't possible to use a [union](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#union)? There should be a way to filter without running several queries at once.

Reply:
@sduenas I managed to apply the filter within the same query without needing a `union`, please check the changes. Thank you!

Reply:
@sduenas Thank you for the pointer to `Subquery` functionality, now I have merged both queries into a single one.